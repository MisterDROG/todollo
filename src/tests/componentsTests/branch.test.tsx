import Branch from "../../components/branch/branch";
import { screen } from '@testing-library/react'
import { renderWithProviders } from '../testUtils/testFunctions';
import userEvent from "@testing-library/user-event";

describe('Branch tests', () => {
    test('Is branch renders right', () => {

        const testingProps = {
            "branchCode": "+house",
            "branchName": "House"
        }

        const { container } = renderWithProviders(<Branch branch={testingProps} />)
        const divBranchContainer = container.querySelector('.branch-container')
        const pBranch = screen.getByText(/House/i)

        expect(divBranchContainer).toHaveStyle({ backgroundColor: '' })
        expect(pBranch).toBeInTheDocument()
        expect(container).toMatchSnapshot()
    })

    test('Is branch inputs right', () => {

        const testingProps = {
            "branchCode": "+house",
            "branchName": "House"
        }

        renderWithProviders(<Branch branch={testingProps} />)
        const inputNewCard = screen.getByPlaceholderText(/New Task.../i)
        expect(inputNewCard).toHaveTextContent('')

        userEvent.type(inputNewCard, 'Test Card')
        expect(screen.getByRole('textbox')).toHaveValue('Test Card')
    })
})