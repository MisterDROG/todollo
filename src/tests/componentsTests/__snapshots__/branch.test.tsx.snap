// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Branch tests Is branch renders right 1`] = `
<div>
  <div
    class="branch"
  >
    <div
      class="branch-container"
    >
      <div
        class="branch__name-container"
      >
        <p
          class="branch__name"
        >
          House
        </p>
        <button
          class="branch__button-delete"
        >
          X
        </button>
      </div>
      <form
        class="branch__form-new-card"
      >
        <input
          class="branch__input-new-card"
          placeholder="New Task..."
          type="text"
          value=""
        />
        <button
          class="branch__button-add-card"
        >
          Add
        </button>
      </form>
    </div>
  </div>
</div>
`;
